metals/src/main/scala/scala/meta/internal/metals/codeactions/RewriteBracesParensCodeAction.scala

```diff
@@ -15,56 +15,71 @@ import scala.meta.tokens.Token
 import org.eclipse.lsp4j.CodeActionParams
 import org.eclipse.{lsp4j => l}
 
+/**
+ * Rewrite parens to brackets and vice versa.
+ * It's a transformation between Term.Apply(_, List(_: Term)) and Term.Apply(_, List(Term.Block(List(_: Term))))
+ * Term.Block is equivalent to "surrounded by braces"
+ *
+ * Parens to brackets scenarios:
+ * 1: def foo(n: Int) = ???
+ *    foo(5)           ->   foo{5}
+ * 2: x.map(a => a)    ->   x.map{a => a}
+ * 3: x.map(_ match {       x.map{_ match {
+ *      case _ => 0    ->     case _ => 0
+ *    })                    }}
+ * Brackets to parens scenarios are the opposite of the ones above.
+ */
 class RewriteBracesParensCodeAction(
     trees: Trees
 ) extends CodeAction {
-
+  import RewriteBracesParensCodeAction._
   override def kind: String = l.CodeActionKind.RefactorRewrite
 
   override def contribute(params: CodeActionParams, token: CancelToken)(implicit
       ec: ExecutionContext
   ): Future[Seq[l.CodeAction]] = Future {
     val path = params.getTextDocument().getUri().toAbsolutePath
     val range = params.getRange()
-    val applyWithSingleFunction: Term.Apply => Boolean = {
-      case Term.Apply(_, List(_: Term.Function)) => true
-      case Term.Apply(_, List(Term.Block(List(_: Term.Function)))) => true
-      case _ => false
-    }
-    trees
-      .findLastEnclosingAt[Term.Apply](
-        path,
-        range.getStart(),
-        applyWithSingleFunction
-      )
+    val applyTree =
+      if (range.getStart == range.getEnd)
```

Comment:
@tgodzik what do you think about this condition? IMO it reduces the ambiguity of scenarios where the user selects multiple lines and sees "rewrite to braces" code action

Reply:
Sound great. Rewrite braces, convert pattern match, is that all which could benefit from that distinction?

Reply:
string interpolation actions also, really anything that has a single title